### ISSUE TITLE
Build error after pull with new dynasm

### ISSUE THREAD

**Issue by lukego** [LuaJIT/LuaJIT#126]
We are seeing a compile error (DynASM version mismatch) if we first compile 55c3b29 and then checkout and compile the latest v2.1 (89982b784). Could be a Makefile bug?

Steps to reproduce:

```
$ make clean
$ git checkout 55c3b29
$ make -j
$ git checkout v2.1
$ make -j
...
HOSTCC    host/buildvm.o
In file included from host/buildvm.c:73:0:
vm_x86.dasc:8:2: error: #error "Version mismatch between DynASM and included encoding engine"
 |.arch x86
  ^
Makefile:682: recipe for target 'host/buildvm.o' failed
make[1]: *** [host/buildvm.o] Error 1
```

**Comment by MikePall** [comment]
Dependency fixed in v2.1 HEAD.

I can't retroactively fix this in the git history without undue consequences. So you'll still have to remember to do a make clean when checking out older commits.

**Comment by lukego** [comment]
Thanks for clarifying!
